### DISCUSSION TITLE
What's reason to mass replacing SCSS variables to CSS variables?

### DISCUSSION THREAD

**Discussion by WinterSilence** [twbs/bootstrap#36035] [Q&A]
I'm seeing a massive replacement of SCSS with CSS variables in components. What caused these changes?

CSS variables doesn't look that impressive in **rendered** CSS.

pros:
- dynamic assign in CSS file
- inline assing in HTML (i.e. bad practice as this called in HTML manuals)

cons:
- slow render, uses more memory/CPU
- harder to debug/support
- less readable SCSS

**Comment by mdo** [comment]
CSS variables offer real-time customization, without the need to recompile Sass. As we continue to iterate on their implementation, we've have eve more global control over the styles, be able to support dynamic (global or component level) dark mode, and more.

**Reply by WinterSilence** [reply]
@mdo
> CSS variables offer real-time customization, without the need to recompile Sass. 

maybe it's `+` in development, but not in production

> dark mode

it can be implemented by classes without any problems

> global or component level

a few global variables - it's fine, but variables in components is bad for performance and optimization (can't extracts critical-path).
